python/pyiceberg/exceptions.py

```diff
@@ -34,3 +38,27 @@ class AlreadyExistsError(Exception):
 
 class ValidationError(Exception):
     """Raises when there is an issue with the schema"""
+
+
+class BadCredentialsError(Exception):
```

Comment:
Should the REST exceptions all inherit from `RESTError`? That gives callers more flexibility to catch REST-related problems in a single `except`.

Reply:
Yes, I like it, and updated this in the code